Tests/WorkspaceTests/TemplateTests.swift

```diff
@@ -95,13 +102,95 @@ class TemplateTests: XCTestCase {
                 let fs = localFileSystem
                 let path = tmpPath.appending(component: "Foo")
                 
-                let result = try SwiftPMProduct.SwiftPackage.executeProcess(["create", "Foo", "--template", "TestTemplate", "--config-path", prefix.pathString], packagePath: tmpPath)
-                let stderr = try result.utf8stderrOutput()
+                let _ = try SwiftPMProduct.SwiftPackage.executeProcess(["create", "Foo", "--template", "ExeTemplate", "--config-path", prefix.pathString], packagePath: tmpPath)
                 
-                XCTAssert(fs.exists(path), stderr)
-                XCTAssert(fs.exists(path.appending(component: "Package.swift")), stderr)
                 XCTAssertBuilds(path)
+                
+                if let readMe = try fs.readFileContents(path.appending(component: "README.md")).validDescription {
+                    XCTAssert(readMe == transformedREADME(with: "Foo"))
+                }
+            }
+        }
+    }
+    
+    func testInitWithDefaultTemplate() throws {
+        try XCTSkipIf(InitPackage.createPackageMode == .legacy)
+        fixture(name: "Templates/MakePackageWithTemplate") { prefix in
+            try testWithTemporaryDirectory { tmpPath in
+                let fs = localFileSystem
+                let path = tmpPath.appending(component: "Bar")
+                try fs.createDirectory(path)
+                
+                let _ = try SwiftPMProduct.SwiftPackage.executeProcess(["init", "--config-path", prefix.pathString], packagePath: path)
+                
+                XCTAssert(fs.exists(path.appending(component: "Package.swift")))
+                XCTAssert(fs.exists(path.appending(components: "src", "main.swift")))
+            }
+        }
+    }
+    
+    func testCreateWithDefaultTemplate() throws {
+        try XCTSkipIf(InitPackage.createPackageMode == .legacy)
+        fixture(name: "Templates/MakePackageWithTemplate") { prefix in
+            try testWithTemporaryDirectory { tmpPath in
+                let fs = localFileSystem
+                let path = tmpPath.appending(component: "Foo")
+                
+                let _ = try SwiftPMProduct.SwiftPackage.executeProcess(["create", "Foo", "--config-path", prefix.pathString], packagePath: tmpPath)
+                
+                XCTAssert(fs.exists(path.appending(component: "Package.swift")))
+                XCTAssert(fs.exists(path.appending(components: "src", "main.swift")))
             }
         }
     }
+    
+    private func transformedManifest(with name: String) -> String {
+        return """
+// swift-tools-version:5.5
```

Comment:
this should be currentToolchain? otherwise it will break next version

Reply:
But that fixtures don't change so the manifest is just left at whatever version it was created with. Changing it here whenever a new version is released would break the tests.

If we wanted to keep the manifest tools version updated than yeah it would be good to use

Reply:
I think you're right, @miggs597, making it the current version would apply new semantics to the code, which could cause a break.  Probably one of the tasks for each release (which we can perhaps automate) will be try upgrade these to latest tools version and to make sure they still build, or make any necessary changes to accommodate the changed semantics of the new tools version.